### DISCUSSION TITLE
logger.info - ParameterizedMessage

### DISCUSSION THREAD

**Discussion by axedrax** [nextgenhealthcare/connect#5343] [Q&A]
Just upgraded from 3.9.0 to 4.1.0 and noticed that when I use logger.info, it displays ParameterizedMessage instead of just the string that I want to output:

[2022-08-10 09:43:37.603] INFO (deploy:?): ParameterizedMessage[messagePattern=testing, stringArgs=[], throwable=null]

Any ideas?

**Comment by pacmano1** [comment]
If you post the same thing here and on the forums, please reference that fact.

**Comment by pacmano1** [comment]
What is the exect logger statement you are using?

**Reply by axedrax** [reply]
In the Deploy script:

logger.info('testing');

**Reply by pacmano1** [reply]
LOL, well that should work as normal assuming you are using a default install.

**Reply by axedrax** [reply]
I'm using the default install.

**Comment by axedrax** [comment]
I'm thinking this has something to do with Mirth 4.1.0 using log4j2 instead of log4j.

**Comment by sj03rs** [comment]
We also have this issue with 4.1.0.
logger.warn and logger.error work fine but logger.info does as stated above.
```
logger.error('This is error'); > [2022-08-11 15:54:37.597]  ERROR  (js-connector:138): This is error
logger.warn('This is warn'); > [2022-08-11 15:54:37.596]  WARN  (js-connector:138): This is warn
logger.info('This is info'); > [2022-08-11 15:54:37.596]  INFO  (js-connector:138): ParameterizedMessage[messagePattern=This is info, stringArgs=[], throwable=null]
```

**Reply by pacmano1** [reply]
Oh, I misread this thread, it does in fact show the contents of the logger.info , just with the extra stuff. 

`logger.info('This is a test')`

returns:

[2022-08-11 08:25:54.704]  INFO  (deploy:138): ParameterizedMessage[messagePattern=**This is a test**, stringArgs=[], throwable=null]

**Comment by axedrax** [comment]
Exactly! I didn't even think about trying .error and .warn.

**Comment by rglenn237** [comment]
https://docs.nextgen.com/bundle/Mirth_User_Guide_41/page/connect/connect/topics/c_Logging_with_JavaScript_connect_ug.html Might have more info for what you are seeing,

**Comment by jonbartels** [comment]
Hmmm https://logging.apache.org/log4j/2.x/manual/messages.html says
> [ParameterizedMessage](https://logging.apache.org/log4j/2.x/log4j-api/apidocs/org/apache/logging/log4j/message/ParameterizedMessage.html) handles messages that contain "{}" in the format to represent replaceable tokens and the replacement parameters.

**Comment by axedrax** [comment]
Appears that the constructor that Mirth is using is deprecated:

![image](https://user-images.githubusercontent.com/110990067/188710968-8bca0f17-6b4c-4000-80af-ee38b41e5879.png)

**Reply by jonbartels** [reply]
Mirth should not use the deprecated form. File an issue for that.

For your current problem - Does calling the deprecated form cause the unusual output from Mirth Connect? It's deprecated but it should still work.

**Comment by axedrax** [comment]
It does cause unusual output. See the post above from sj03rs on 08/11/22. It's the best description of the 3 types of logger outputs.

**Comment by matthydehoog** [comment]
Even stranger, Using

logger.info("2 User {} has logged in using id {}"); in a source transformer gives:

[2022-09-14 17:04:10.198] INFO (transformer:?): User {} has logged in using id {}

Using the same code in a channel deploy script gives:

[2022-09-14 17:04:56.555] INFO (deploy:?): ParameterizedMessage[messagePattern=2 User {} has logged in using id {}, stringArgs=[], throwable=null]

MC 4.1.1

**Comment by jonbartels** [comment]
https://github.com/nextgenhealthcare/connect/issues/5423

**Comment by robdejournett** [comment]
Also this is not seen in the transformer, but is seen in the connector.   Lots of strange behavior. 

In the transform this works as expected:  logger.info("foo");  
INFO (transformer:?): foo

In the connector:  logger.info("foo");
INFO (js-connector:?): ParamaterizedMessage .....

**Comment by anei20018cool** [comment]
open  ArrayAppender.java  line 43  ,  change  logEvent.getMessage() to logEvent.getMessage().getFormattedMessage()

**Reply by axedrax** [reply]
Can't seem to find ArrayAppender.java. Where is your file located?

**Reply by anei20018cool** [reply]
Source code path  ： server/src/com/mirth/connect/plugins/serverlog/ArrayAppender.java
